packages/sdk/test-next/messageStatus.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { describe, expect, it } from 'vitest'
+
+import { CrossChainMessenger, MessageStatus } from '../src'
+import { l1Provider, l2Provider } from './testUtils/ethersProviders'
+
+const crossChainMessenger = new CrossChainMessenger({
+  l1SignerOrProvider: l1Provider,
+  l2SignerOrProvider: l2Provider,
+  l1ChainId: 5,
+  l2ChainId: 420,
+  bedrock: true,
+})
+
+describe('getMessageStatus', () => {
+  it(`should be able to correctly find a finalized withdrawal`, async () => {
+    /**
+     * Tx hash of a withdrawal
+     *
+     * @see https://goerli-optimism.etherscan.io/tx/0x8fb235a61079f3fa87da66e78c9da075281bc4ba5f1af4b95197dd9480e03bb5
+     */
+    const txWithdrawalHash =
+      '0x8fb235a61079f3fa87da66e78c9da075281bc4ba5f1af4b95197dd9480e03bb5'
+
+    const txReceipt = await l2Provider.getTransactionReceipt(txWithdrawalHash)
+
+    expect(txReceipt).toBeDefined()
+
+    expect(await crossChainMessenger.getMessageStatus(txWithdrawalHash, 0, 9370789 - 1000, 9370789)).toBe(
+      MessageStatus.RELAYED
+    )
+  }, 20_000)
+
+  it(`should return READY_FOR_RELAY if not in block range`, async () => {
+    const txWithdrawalHash =
+      '0x8fb235a61079f3fa87da66e78c9da075281bc4ba5f1af4b95197dd9480e03bb5'
+
+    const txReceipt = await l2Provider.getTransactionReceipt(txWithdrawalHash)
+
+    expect(txReceipt).toBeDefined()
+
+    expect(
+      await crossChainMessenger.getMessageStatus(txWithdrawalHash, 0, 0, 0)
+    ).toBe(MessageStatus.READY_FOR_RELAY)
```

Comment:
This is READY_FOR_RELAY because, though the query filter failed, we found the proved withdrawal. Kinda odd so not sure if we want to change behavior here https://github.com/ethereum-optimism/optimism/blob/develop/packages/sdk/src/cross-chain-messenger.ts#L729

Reply:
I think this is correct.  Within this block range it is READY_FOR_RELAY so even though it's slightly confusing it feels correct to me. You could imagine using the filters to debug the state of a tx in a block-range even though you know it's state changed more recently.

It's not perfect though because it can still return the wrong state if the event happened BEFORE not after your blockrange but we will deal with that when we add better documentation

Reply:
actually within the given block range it doesn't even exist 😬 , but up to you what to resolve here vs. follow up